diagnostic.md

```diff
@@ -6,70 +6,71 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+get '/books/4' => 'books#show'
+resources :books, only: [:show]
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+config/routes.rb
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+Prior to running this code you have created a migration that will alter the database in some way.  The above code will execute the migration so that the database will actually be changed and the schema updated.
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+The generate scaffold command is a superset of the generate controller command.  Generate scaffolding will generate a model file with some specs, a route, some serializers, and a migration in addition to a controller.
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+It allows us to customize the output that rails sends as JSON from our server.  It's taking the place of the View in the MVC paradigm.
 ```
 
 What do we expect the command `Patient.all` to return?
 
 ```md
-// your response here
+Every instance that has been created off of the class Patient.
 ```
 
 We use the constant `Patient` in the `PatientsController`, but where is it
 actually defined?
 
 ```md
-// your response here
+The Model.
 ```
 
 Why do our `model` files appear to be lacking so much code, when they're doing
 so much for us?
 
 ```md
-// your response here
+Becuase they are inheriting hundreds of class and instance methods from Application Record and Object.
 ```
 
 What does typing `bin/rails dbconsole` in a directory that houses a rails app do for
 us?
 
 ```md
-// your response here
+Figures out which database you're using and drops you into whichever command line interface you would use with it.
```

Comment:
This allows you to interact with your Rails app, whereas the next command will actually take you into the underlying database